mutable-containers/src/Data/Mutable/Class.hs

```diff
@@ -2,6 +2,7 @@
 {-# LANGUAGE FlexibleContexts  #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE TypeFamilies      #-}
+{-# LANGUAGE TypeOperators #-}
```

Comment:
@gregwebs Do you see any problems using `TypeOperators` in `mutable-containers`? `mono-traversable` was already using the extension and [ghc has had this since 6.8.1](https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/type_operators.html?highlight=xtypeoperators#extension-TypeOperators)

Reply:
can you link to information about what error this is avoiding?

Reply:
```text
src\Data\Builder.hs:71:13: warning: [GHC-58520] [-Wtype-equality-requires-operators]
Warning:     The use of ‘~’ without TypeOperators
    will become an error in a future GHC release.
    Suggested fix: Perhaps you intended to use TypeOperators.
   | instance (a ~ Char) => ToBuilder [a] TB.Builder 
```

[-Wtype-equality-requires-operators](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/using-warnings.html?highlight=wtype%20equality%20requires%20operators#ghc-flag--Wtype-equality-requires-operators)
[proposal 0371](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0371-non-magical-eq.md)